src/test/ui/nll/issue-52113.stderr

```diff
@@ -1,11 +1,12 @@
-error[E0623]: lifetime mismatch
+error: unsatisfied lifetime constraints
   --> $DIR/issue-52113.rs:43:9
    |
 LL | fn produce_err<'a, 'b: 'a>(data: &'b mut Vec<&'b u32>, value: &'a u32) -> impl Bazinga + 'b {
-   |                                  --------------------         ------- these two types are declared with different lifetimes...
-LL |     let x = move || { //~ ERROR lifetime mismatch
-   |         ^ ...but data from `value` flows into `data` here
+   |                --  -- lifetime `'b` defined here
+   |                |
+   |                lifetime `'a` defined here
```

Comment:
I think we should file a bug here too -- in general, it's often dissatisfying to highlight the *declaration*. To do better, though, we'd presumably have to look at the MIR and see if we can identify the parameter. (You could imagine checking if the lifetime only appears in one parameter's type and using that to guess that we should highlight the parameter type, also)

Reply:
[issue-52663-lifetimes-not-included-in-span](https://rust-lang.zulipchat.com/#narrow/stream/122657-wg-nll/topic/issue-52663-lifetimes-not-included-in-span) in Zulip and [this comment](https://github.com/rust-lang/rust/issues/52663#issuecomment-409036656) discussed this briefly as a potential part of #52973.